doc/api/animation_api.rst

```diff
@@ -123,8 +124,37 @@ artist at a global scope and let Python sort things out.  For example ::
                        init_func=init, blit=True)
    plt.show()
 
-The second method is to use `functools.partial` to 'bind' artists to
-function.  A third method is to use closures to build up the required
+The second method is to use `functools.partial` to pass arguments to the
+function. ::
+
+   import numpy as np
+   import matplotlib.pyplot as plt
+   from matplotlib.animation import FuncAnimation
+   from functools import partial
+
+   fig, ax = plt.subplots()
+   ln, = plt.plot([], [], 'ro')
+
+   def init():
+      ax.set_xlim(0, 2*np.pi)
+      ax.set_ylim(-1, 1)
+      return ln,
+
+   def update(frame, x, y):
```

Comment:
This is just wrong isn't it?  you are appending to x and y and then setting the data with xdata and ydata.  I guess I am never clear if x and y are references or not, but at the very least this is confusing...

Reply:
I thought the same, hence the use of `offset`, should I revert to that instead?

Reply:
I don't know.  I'd not do it this way, so whoever would do it this way should write the doc ;-)

Reply:
@tacaswell Suggested the example, would this be better instead : 
```py
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.animation import FuncAnimation
from functools import partial

fig, ax = plt.subplots()
ln, = plt.plot([], [], 'ro')

def init():
   ax.set_xlim(0, 2*np.pi)
   ax.set_ylim(-1, 1)
   return ln,

def update(frame, x_offset=0, y_offset=0):
   xdata.append(frame+x_offset)
   ydata.append(np.sin(frame)+y_offset)
   ln.set_data(xdata, ydata)
   return ln,

x_offset = 2
y_offset = 0.2
ani = FuncAnimation(
fig, partial(update, x_offset=x_offset, y_offset=y_offset),
frames=np.linspace(0, 2 * np.pi, 128),
init_func=init, blit=True)

plt.show()
```

Reply:
I mean the above looks better to me.  But I don't understand, at all, why we wouldn't just use globals here.  If an example where this is practically _better_ than just using a global, we should write such an example.